src/executor/cursor.rs

```diff
@@ -981,17 +1592,106 @@ mod tests {
+                [4_u64.into()]
+            ),])]
+        );
+
+        Ok(())
+    }
+
+    #[smol_potat::test]
+    async fn composite_keys_4_desc() -> Result<(), DbErr> {
+        use xyz_entity::*;
+
+        let db = MockDatabase::new(DbBackend::Postgres)
+            .append_query_results([[Model {
+                x: 'x' as i32,
+                y: "y".into(),
+                z: 'z' as i64,
+            }]])
+            .into_connection();
+
+        assert!(!Entity::find()
+            .cursor_by((Column::X, Column::Y, Column::Z))
+            .last(4)
+            .desc()
+            .all(&db)
+            .await?
+            .is_empty());
+
+        assert_eq!(
+            db.into_transaction_log(),
+            [Transaction::many([Statement::from_sql_and_values(
+                DbBackend::Postgres,
+                [
+                    r#"SELECT "m"."x", "m"."y", "m"."z""#,
+                    r#"FROM "m""#,
+                    r#"ORDER BY "m"."x" ASC, "m"."y" ASC, "m"."z" ASC"#,
+                    r#"LIMIT $1"#,
+                ]
+                .join(" ")
+                .as_str(),
+                [4_u64.into()]
+            ),])]
+        );
+
+        Ok(())
+    }
+
+    #[smol_potat::test]
+    async fn composite_keys_5() -> Result<(), DbErr> {
+        use xyz_entity::*;
+
+        let db = MockDatabase::new(DbBackend::Postgres)
+            .append_query_results([[Model {
+                x: 'x' as i32,
+                y: "y".into(),
+                z: 'z' as i64,
+            }]])
+            .into_connection();
+
+        assert!(!Entity::find()
+            .cursor_by((Column::X, Column::Y, Column::Z))
+            .after(('x' as i32, "y".to_owned(), 'z' as i64))
+            .first(4)
+            .all(&db)
+            .await?
+            .is_empty());
+
+        assert_eq!(
+            db.into_transaction_log(),
+            [Transaction::many([Statement::from_sql_and_values(
+                DbBackend::Postgres,
+                [
+                    r#"SELECT "m"."x", "m"."y", "m"."z""#,
+                    r#"FROM "m""#,
+                    r#"WHERE ("m"."x" = $1 AND "m"."y" = $2 AND "m"."z" > $3)"#,
+                    r#"OR ("m"."x" = $4 AND "m"."y" > $5)"#,
+                    r#"OR "m"."x" > $6"#,
+                    r#"ORDER BY "m"."x" ASC, "m"."y" ASC, "m"."z" ASC"#,
+                    r#"LIMIT $7"#,
+                ]
+                .join(" ")
+                .as_str(),
+                [
+                    ('x' as i32).into(),
+                    "y".into(),
+                    ('z' as i64).into(),
+                    ('x' as i32).into(),
+                    "y".into(),
+                    ('x' as i32).into(),
+                    4_u64.into(),
+                ]
             ),])]
         );
 
         Ok(())
     }
 
     #[smol_potat::test]
-    async fn composite_keys_5() -> Result<(), DbErr> {
+    async fn composite_keys_5_desc() -> Result<(), DbErr> {
```

Comment:
We should avoid changing existing test cases, since it shrinks our test coverage. Can you extract the common parts and make it a new test case?

Reply:
I didn't change any existing tests, but duplicated all existing ones with `desc` order, it shows a bit incorrectly by github, I assume because names `composite_keys_5` and `composite_keys_5_desc` are similar

Reply:
Oh thanks!